libsrc/interface/readuser.cpp

```diff
@@ -105,20 +110,47 @@ namespace netgen
                 while (1)
                   {
                     int pi, hi;
-                    Point<3> p;
+		    Point<3> p;
+		    string p1tmp, p2tmp, p3tmp;
 
                     in >> pi;
                     if (pi == -1)
                       break;
 		    
                     in >> hi >> hi >> hi;
-                    in >> p(0) >> p(1) >> p(2);
-
-                    mesh.AddPoint (p);
+		    if (DnotationSet == false) {
+			in >> p1tmp >> p2tmp >> p3tmp;
+			if (p1tmp.find("D") != std::string::npos){
+			    DnotationSet = true;
+			    Dnotation = true;
+			    cout << "Attention: in your UNV file, D is used as an exponent prefix instead of E" << endl;
+			    std::replace( p1tmp.begin(), p1tmp.end(), 'D', 'E');
+			    std::replace( p2tmp.begin(), p2tmp.end(), 'D', 'E');
+			    std::replace( p3tmp.begin(), p3tmp.end(), 'D', 'E');
+			}
+			p(0) = std::stod(p1tmp);
+			p(1) = std::stod(p2tmp);
+			p(2) = std::stod(p3tmp);
+		    }
+		    else if (Dnotation == true) {
+		        in >> p1tmp >> p2tmp >> p3tmp;
+			std::replace( p1tmp.begin(), p1tmp.end(), 'D', 'E');
+			std::replace( p2tmp.begin(), p2tmp.end(), 'D', 'E');
+			std::replace( p3tmp.begin(), p3tmp.end(), 'D', 'E');
+			p(0) = std::stod(p1tmp);
+			p(1) = std::stod(p2tmp);
+			p(2) = std::stod(p3tmp);
+		    }
+		    else{
+			in >> p(0) >> p(1) >> p(2);
+		    }
+		    mesh.AddPoint(p);
                   }
 		cout << "read " << mesh.GetNP() << " points" << endl;
                 Point3d pmin, pmax;
+		cout << "Get Box" << endl;
                 mesh.GetBox (pmin, pmax);
+		cout << "Pmin: " << pmin << " Pmax: " << pmax << endl;
                 if(fabs(pmin.Z() - pmax.Z()) < 1e-10 * Dist(pmin, pmax))
```

Comment:
Can you add message level to the cout? So that one can disable the output. It is missing there in the other cout statements as well. You can set it just by adding << IM(level) after the cout, I think 3-5 are appropriate message levels depending on importance.
```
cout << IM(5) << ...
```

Reply:
Done